src/build.d

```diff
@@ -913,6 +916,61 @@ auto getHostDMDPath(string hostDmd)
         static assert(false, "Unrecognized or unsupported OS.");
 }
 
+version(Windows)
+{
+    /*
+    Gets the absolute path to the host's vshwere executable
+
+    Params:
+        outputFolder = this build's output folder
+    Returns: a string that is the absolute path of the host's vswhere executable
+    */
+    auto getHostVSWhere(string outputFolder)
+    {
+        // Check if vswhere.exe can be found in the host's PATH
+        const where = ["where", "vswhere"].execute;
+        if (where.status == 0)
+            return where.output;
+
+        // Check if vswhere.exe is in the standard location
+        auto standardPath = ["cmd", "/C", "echo", `%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe`].execute.output;
+        standardPath = standardPath.replace(`"`, "");    // Remove quotes surrounding the path
+        standardPath = standardPath.replace("\r\n", ""); // Remove trailing newline characters
+        if (standardPath.exists)
+            return standardPath;
+
+        // Check if it has already been dowloaded to this build's output folder
+        const outputPath = outputFolder.buildPath("vswhere").exeName;
+        if (outputPath.exists)
+            return outputPath;
+
+        // try to download it
+        immutable url = "https://github.com/Microsoft/vswhere/releases/download/2.5.2/vswhere.exe";
+        enum tries = 3;
+        foreach(i; 0..tries)
+        {
+            import std.net.curl : download, HTTPStatusException;
+            try
+            {
+                log("Downloading %s ...", url);
+                download(url, outputPath);
+                return outputPath;
+            }
+            catch(HTTPStatusException e) 
+            {
+                if (e.status == 404) throw e;
+                else 
+                {
+                    log("Failed to download %s (Attempt %s of %s)", url, i + 1, tries);
+                    continue;
+                }
+            }
+        }
```

Comment:
Better wrap this in a function as we can reuse it for a cross-platform download of the bootstrap compiler too, then.

Reply:
Done.  I didn't change the existing bootstrap compiler download to avoid introducing issues.  I can always do that as a followup.